pkg/controllers/vdb/readonlyonlineupgrade_reconciler.go

```diff
@@ -454,17 +457,59 @@ func (o *ReadOnlyOnlineUpgradeReconciler) drainSubcluster(ctx context.Context, s
 		return ctrl.Result{}, err
 	}
 
-	if img != o.Vdb.Spec.Image {
-		scName := sts.Labels[vmeta.SubclusterNameLabel]
-		o.Log.Info("rerouting client traffic from subcluster", "name", scName)
-		if err := o.routeClientTraffic(ctx, scName, true); err != nil {
-			return ctrl.Result{}, err
-		}
+	if img == o.Vdb.Spec.Image {
+		return ctrl.Result{}, nil
+	}
+	scName := sts.Labels[vmeta.SubclusterNameLabel]
+	o.Log.Info("rerouting client traffic from subcluster", "name", scName)
+	if err := o.routeClientTraffic(ctx, scName, true); err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	o.Log.Info("starting check for active connections in subcluster", "name", scName)
+	return o.runSubclusterDrain(ctx, scName)
+}
+
+func (o *ReadOnlyOnlineUpgradeReconciler) runSubclusterDrain(ctx context.Context, scName string) (ctrl.Result, error) {
+	drainSubclusterAnnotation := "vertica.com/drain-start-time-" + scName
```

Comment:
Add this to annotations.yaml, the prefix will be a constant and we will use a function to build the final annotation.

You should also check early if there is already an annotation with prefix "vertica.com/drain-start-time-" but a different subcluster. That would mean we are already draining a draining a subcluster which is not the current one so we should just return.